astoria/consumers/astwifid.py

```diff
@@ -0,0 +1,164 @@
+            except JSONDecodeError:
+                LOGGER.warning("Received bad JSON in metadata manager message.")
+        else:
+            LOGGER.warning("Received empty metadata manager message.")
+
+    async def handle_metadata(self, metadata: Metadata) -> None:
+
+        if self._lifecycle:
+            if metadata.wifi_enabled:
+                # Check if creds match
+                pass
+            else:
+                # Turn it off!
+                await self._lifecycle.stop_hotspot()
+                self._lifecycle = None
+        else:
+            if metadata.wifi_enabled:
+                # Turn it on!
+                self._lifecycle = WiFiHotspotLifeCycle(
+                    metadata.wifi_ssid,
+                    metadata.wifi_psk,
+                    metadata.wifi_region,
+                    metadata.wifi_interface,
+                )
+                asyncio.ensure_future(self._lifecycle.run_hotspot())
+
+
+class WiFiHotspotLifeCycle:
+    ssid: str
+    psk: str
+    region: str
+    interface: str
+    config_file_path: Path = Path("/tmp/astwifid_hostapd_config")
+    proc: Optional[Popen] = None
+    running = True
+
+    def __init__(self, ssid: str, psk: str, region: str, interface: str) -> None:
+        print("Starting WiFi Hotspot lifecycle")
+        self.ssid = ssid
+        self.psk = psk
+        self.region = region
+        self.interface = interface
+
+    async def run_hotspot(self) -> None:
+        self.running = True
+        LOGGER.info(f"Starting WiFi Hotspot \"{self.ssid}\" on {self.interface}")
+        self.gen_hostapd_config()
+        self.proc = Popen(["hostapd", self.config_file_path])
+
+        while self.running:
+            sc = self.proc.poll()
+            if sc is None:
+                continue
+
+            LOGGER.info(f"hostapd terminated with status code {sc}, restarting...")
+            self.proc = Popen(["hostapd", self.config_file_path])
+            LOGGER.info(f"hostapd started wth PID: {self.proc.pid}")
+
+
+    def gen_hostapd_config(self):
+        self.config_file_path = Path(check_output(["mktemp"]).decode().strip())
+        LOGGER.debug(f"Writing hostapd configuration to {self.config_file_path}")
+        config = {
+            "interface": self.interface,
+            # "bridge": "br0",
+            "ssid": self.ssid,
+            # "driver": "nl80211",
+            "country_code": self.region,
+            "channel": 7,
+            "hw_mode": "g",
+            # Bit field: bit0 = WPA, bit1 = WPA2
+            "wpa": 2,
+            # Bit field: 1=wpa, 2=wep, 3=both
+            "auth_algs": 1,
+            # Set of accepted cipher suites; disabling insecure TKIP
+            "wpa_pairwise": "CCMP",
+            # Set of accepted key management algorithms
+            "wpa_key_mgmt": "WPA-PSK",
+            "wpa_passphrase": self.psk,
+        }
+        contents = "\n".join([f"{k}={config.get(k)}" for k in config])
```

Comment:
You can just give join an iterable here.
```suggestion
        contents = "\n".join(f"{k}={config.get(k)}" for k in config)
```

Or we could potentially be a bit more clever...

```suggestion
        contents = "\n".join(f"{key}={value}" for key, value in config.items())
```